### PR TITLE
fix(eval-hub): use configured primary metric for benchmark score display

### PR DESCRIPTION
**Pull request by nickmazzi**
https://issues.redhat.com/browse/RHOAIENG-89019

## Description

`formatBenchmarkScore` now accepts an optional `primaryMetric` parameter sourced from the benchmark config's `primary_score.metric` field. Previously, score display only checked `test.primary_score` and then fell back to the well-known metrics `acc_norm` and `acc`. Benchmarks from providers like garak (whose primary metric is `attack_success_rate`) or custom providers with non-standard metric names would show `-` instead of their actual score.

Changes:
- `formatBenchmarkScore` resolution order is now: `test.primary_score` > configured `primaryMetric` lookup > well-known fallback list (`acc_norm`, `acc`, `attack_success_rate`).
- `getResultScore` and `getBenchmarkResultScore` now cross-reference the result benchmark with the job's configured benchmarks to pass through the `primary_score.metric` value.
- `attack_success_rate` added to the well-known fallback list so garak benchmarks without a `test` object still display a score.

image

## How Has This Been Tested?

Unit tests added and run via Jest covering the new logic paths in `formatBenchmarkScore`, `getResultScore`, and `getBenchmarkResultScore`.

Manually tested by hardcoding the GET responses to include garak jobs with 0s.
Use this patch to apply the same test
[garak-hardcoded-test-data.patch](https://github.com/user-attachments/files/31519831/garak-hardcoded-test-data.patch)

## Test Impact

Added and modified tests in `evaluationUtils.spec.ts`:
- **`formatBenchmarkScore`** — new describe block with 7 test cases covering: `test.primary_score` priority, `primaryMetric` parameter usage, `acc_norm`/`acc` fallback, `attack_success_rate` fallback, null returns for unrecognized/missing metrics, and non-numeric metric skipping.
- **`getResultScore`** — 3 new test cases for garak `attack_success_rate` fallback (zero and non-zero values) and configured `primary_score.metric` from benchmark config.
- **`getBenchmarkResultScore`** — 2 new test cases for garak `attack_success_rate` fallback and configured `primary_score.metric` from benchmark config.

## Request review criteria:

Self checklist (all need to be checked):
- [ ] The developer has manually tested the changes and verified that the changes work
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)
- [x] The code follows our [Best Practices](/docs/best-practices.md) (React coding standards, PatternFly usage, performance considerations)

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved benchmark score display by prioritizing explicitly configured primary scores.
  * Added support for benchmark-specific primary metrics when calculating result scores.
  * Expanded score fallbacks to include accuracy, normalized accuracy, and attack success rate metrics.
  * Ensured evaluation and benchmark result views use the appropriate configured metric for more consistent scoring.
  * Improved score resolution when benchmarks share duplicate identifiers.

### CODE REVIEW

**Comment by nickmazzi** [comment]
@coderabbitai review

File: packages/eval-hub/frontend/src/app/utilities/evaluationUtils.ts [hunk 1]
```diff
@@ -102,7 +128,9 @@ export const getResultScore = (job: EvaluationJob): string => {
     return '-';
   }
   if (job.results.benchmarks?.length) {
-    return formatBenchmarkScore(job.results.benchmarks[0]) ?? '-';
+    const resultBenchmark = job.results.benchmarks[0];
+    const configBenchmark = getJobBenchmarks(job).find((b) => b.id === resultBenchmark.id);
+    return formatBenchmarkScore(resultBenchmark, configBenchmark?.primary_score?.metric) ?? '-';
```
[the remark this answers is not in the document]
Comment (nickmazzi) [reply packages/eval-hub/frontend/src/app/utilities/evaluationUtils.ts:136]: Addressed via https://github.com/opendatahub-io/odh-dashboard/pull/9511/commits/cc3d437361227fc9a2bf50695657e384612307ce

File: packages/eval-hub/frontend/src/app/utilities/evaluationUtils.ts [hunk 2]
```diff
@@ -102,7 +128,9 @@ export const getResultScore = (job: EvaluationJob): string => {
     return '-';
   }
   if (job.results.benchmarks?.length) {
-    return formatBenchmarkScore(job.results.benchmarks[0]) ?? '-';
+    const resultBenchmark = job.results.benchmarks[0];
+    const configBenchmark = getJobBenchmarks(job).find((b) => b.id === resultBenchmark.id);
```
[the remark this answers is not in the document]
Comment (nickmazzi) [reply packages/eval-hub/frontend/src/app/utilities/evaluationUtils.ts]: Addressed via https://github.com/opendatahub-io/odh-dashboard/pull/9511/commits/cc3d437361227fc9a2bf50695657e384612307ce

**Comment by jefho-rh** [comment]
Thanks for the fix here @nickmazzi!

/lgtm
/approve

## 👤 Manual Testing

image

---

## AI Code Review 🤖 🔍

### 🎯 Verdict: APPROVED ✅

Targeted and correct fix for benchmark score display. The three-tier resolution order (`test.primary_score` → configured `primaryMetric` → well-known fallback list) is sound, the duplicate-ID matching via `benchmark_index` with positional fallback is handled consistently across both `getResultScore` and `getBenchmarkResultScore`, and the garak zero-score edge case is properly covered.

> 💡 The updated test description `'should return dash when metrics has no recognized metric names'` (previously `'should return dash when metrics has neither acc_norm nor acc'`) is a clean example of keeping test names in sync with behavioral changes — this kind of discipline prevents stale spec names from misleading future readers.

---

### 📋 Findings

No actionable findings.

---

### 🧪 Test Verification

| Check | Status | Details |
|-------|--------|---------|
| Lint | ✅ Pass | Clean |
| Type Check | ✅ Pass | Clean |
| Unit Tests | ✅ Pass | 83/83 passed |
| BFF Tests | ➖ N/A | No Go files changed |
| Contract Tests | ➖ N/A | No BFF changes |
